### ISSUE TITLE
x/build/cmd/coordinator: many repeating js-wasm builds running every 15 seconds

### ISSUE THREAD

**Issue by dmitshur** [golang/go#27012]
Over at https://farmer.golang.org/#completed, I'm seeing a large number of js-wasm builders start every 15 seconds for the exact same commit:

![image](https://user-images.githubusercontent.com/1924134/44175241-92f63800-a0b3-11e8-8b50-5930188810f4.png)

Each one completes within a second with the following output:

```
  builder: js-wasm
      rev: dea961ebd9f871b39b3bdaab32f952037f28cd71
 buildlet: (nil *buildlet.Client)
  started: 2018-08-15 21:48:11.498031687 +0000 UTC m=+521730.873199532
    ended: 2018-08-15 21:48:11.658471517 +0000 UTC m=+521731.033639405
  success: true

Events:
  2018-08-15T21:48:11Z ask_maintner_has_ancestor 
  2018-08-15T21:48:11Z finish_ask_maintner_has_ancestor after 24.5ms
  2018-08-15T21:48:11Z skipped_build_missing_dep 

Build log:
skipping build; commit dea961ebd9f871b39b3bdaab32f952037f28cd71 lacks ancestor 3dced519cbabc213df369d9112206986e62687fa
```

This seems unintentional and suboptimal, right? If so, I'd like to better understand why it happens and try to fix it.

/cc @bradfitz

**Comment by dmitshur** [comment]
Not sure how relevant this is, but both commits exist:

- https://github.com/golang/go/commit/dea961ebd9f871b39b3bdaab32f952037f28cd71
- https://github.com/golang/go/commit/3dced519cbabc213df369d9112206986e62687fa

And there are 1735 commits [between](https://github.com/golang/go/compare/dea961ebd9f871b39b3bdaab32f952037f28cd71...3dced519cbabc213df369d9112206986e62687fa) them.

**Comment by bradfitz** [comment]
Couple things:
* this isn't great, but isn't terrible: we're stopping before creating a VM at least.
* we shouldn't be trying to do js-wasm builds for release-branch.go1.10 or many other old branches
* we should keep a global cache `wasSkippedBuild[builderRev{builder, rev}] = true` once discovered, to avoid enqueuing that duplicate work in findWork* later.

**Comment by neelance** [comment]
@dmitshur Is this still an issue?

**Comment by dmitshur** [comment]
It does not appear to be. I've checked now, things look good, and I have not observed other occurrences in a long time.

"we shouldn't be trying to do js-wasm builds for release-branch.go1.10 or many other old branches" is resolved.

I don't think "a global `wasSkippedBuild` cache" was implemented. But it doesn't seem to be very relevant now.

Let's close this as resolved. If we find a problem, we can file a new issue with with current and actionable information.
